Add ADR about carrying applications over to the next cycle

duncanjbrown opened the pull request:
## Context

Writing up decisions following a meeting on 12th August about #2681.

## Changes proposed in this pull request

Add an ADR about what we decided.

## Guidance to review

- Does this fairly reflect the meeting?
- Did we miss anything?

## Link to Trello card

N/A

## Things to check

- [X] This code does not rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

tijmenb commented on adr/0015-carrying-over-applications.md:
```diff
@@ -0,0 +1,81 @@
+  before the application was carried over will need to be cancelled.
+- The applications that were carried over will remain in the database without
+  any further state change
+- Applications which were not yet sent to the provider at end of cycle and also
+  not carried over will still be in the database — we would like to mark these
+  with a new state equivalent to “incomplete at end of cycle”. This state would
+  never be visible to providers.
```
What do we do with applications in awaiting references state? Do we want to cancel these references?

duncanjbrown replied:
See two bullet points up (unless I've missed something)

mnacos commented on adr/0015-carrying-over-applications.md:
```diff
@@ -0,0 +1,81 @@
+
+#### After the application can’t progress any further
+
+- Conditions met (recruited)
```
`Conditions met (recruited)` appears both here and in the 'does not make sense' section. Which one should stay?

mnacos commented on adr/0015-carrying-over-applications.md:
```diff
@@ -0,0 +1,81 @@
+#### When the application already has an offer in flight
+
+- Offer
+- Meeting conditions
```
What is 'Meeting conditions'? Is it a future state (some conditions have been met while others are pending)?

mnacos commented on adr/0015-carrying-over-applications.md:
```diff
@@ -0,0 +1,81 @@
+
+#### Before the application reaches the provider
+
+- Withdrawn
```
Is this so that the candidate can un-withdraw and apply again?

duncanjbrown commented:
Thanks for your comments @mnacos! I've removed "Conditions met" from the "makes sense to carry over" section and added some additional text explaining the rationale for when things are carry-over-able.

duncanjbrown commented:
re application choices being in mixed states, I think the rule of thumb is that if at least one choice is not yet "finished" it is not possible to carry over. Once they're all done for, it is.

mnacos commented on adr/0015-carrying-over-applications.md:
```diff
@@ -0,0 +1,87 @@
+
+The current approach for moving applications into Apply again is to copy the
+entire application (including references) and invite the user to add a new
+course choice. This approach seems like it will work here too, with a couple of
```
This is a great addition, thanks!

mnacos reviewed (approved):
Explaining the reasoning behind which states will are carried over makes this document clearer :rocket:
